helm-system-packages-dnf.el

```diff
@@ -0,0 +1,135 @@
+;;; helm-system-packages-dnf.el --- Helm UI for RPM-based distros using dnf. -*- lexical-binding: t -*-
+
+;; Copyright (C) 2018 Damien Cassou <damien@cassou.me>
+
+;; Author: Damien Cassou <damien@cassou.me>
+;; URL: https://github.com/emacs-helm/helm-system-packages
+;; Version: 1.10.1
+;; Package-Requires: ((emacs "24.4") (helm "2.8.6"))
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Helm UI for RPM-based distributions using dnf.
+
+;;; Code:
+(require 'helm-system-packages)
+
+(defcustom helm-system-packages-dnf-actions
+  '(("Show package(s)" . helm-system-packages-dnf-info))
+  "Actions for Helm guix."
+  :group 'helm-system-packages
+  :type '(alist :key-type string :value-type function))
+
+(defun helm-system-packages-dnf--package-attributes-to-org-description-list ()
+  "Convert \"Key   : value\" to \"- Key :: value\".
+This is org format for description list items."
+  (goto-char (point-min))
+  (while (re-search-forward "^\\(\\<[^ ]*\\) +: \\(.*\\)$" nil t)
+    (let ((key (match-string-no-properties 1))
+          (value (match-string-no-properties 2)))
+      (replace-match (format "- %s :: %s" key value) nil t))))
+
+(defun helm-system-packages-dnf--info (packages)
+  "Return a list of (NAME . DESC) describing PACKAGES."
+  (save-match-data
+    (with-temp-buffer
+      (apply #'process-file "dnf" nil t nil "info" packages)
+      (goto-char (point-min))
+      (helm-system-packages-dnf--merge-descriptions)
+      (helm-system-packages-dnf--package-attributes-to-org-description-list)
+      (goto-char (point-min))
+      (cl-loop
+       while (re-search-forward "^- Name :: \\(.*\\)$" nil t)
+       collect (cons (match-string-no-properties 1)
+                     (buffer-substring-no-properties
+                      (match-beginning 0)
+                      (save-match-data
+                        (re-search-forward "^$")
+                        (match-beginning 0))))))))
+
+(defun helm-system-packages-dnf-info (candidate)
+  "Print information about helm CANDIDATE.
+With prefix argument, insert the output at point.
+Otherwise display in `helm-system-packages-buffer'."
+  (helm-system-packages-show-information
+   `((whatever . ,(helm-system-packages-dnf--info
```

Comment:
What is `whatever`?  I think you want to use `uninstalled` for packages without categories.

Reply:
The docstring of `helm-system-packages-show-information` contains

> DESC-ALIST's keys are ignored

I thought that if it is ignored, I can put whatever :-). Can you please fix the docstring?

Also, I'm wondering what is this `uninstalled`. These packages could be installed, I don't know.